x/evm/keeper/grpc_query.go

```diff
@@ -73,66 +76,66 @@ func (q Querier) BurnerInfo(c context.Context, req *types.BurnerInfoRequest) (*t
 	return nil, status.Error(codes.NotFound, "unknown address")
 }
 
-func getExecuteDataAndSigs(ctx sdk.Context, s types.Signer, commandBatch types.CommandBatch) ([]byte, []types.Signature, error) {
-	id := hex.EncodeToString(commandBatch.GetID())
-	sig, sigStatus := s.GetSig(ctx, id)
-	if sigStatus != tss.SigStatus_Signed {
-		return nil, nil, fmt.Errorf("could not find a corresponding signature for sig ID %s", id)
-	}
-
-	switch sig := sig.GetSig().(type) {
-	case *tss.Signature_SingleSig_:
-		commandBatchSig, err := getCommandBatchSig(sig.SingleSig.SigKeyPair, commandBatch.GetSigHash())
-		if err != nil {
-			return nil, nil, err
+func optimizeSignatureSet(sigs map[string]multisigtypes.Signature, addressWeights map[string]sdk.Uint, threshold sdk.Uint) [][]byte {
+	addresses := maps.Keys(addressWeights)
+	sort.SliceStable(addresses, func(i, j int) bool {
+		return addressWeights[addresses[i]].GT(addressWeights[addresses[j]])
+	})
+
+	var optimizedSig [][]byte
+	cumWeight := sdk.ZeroUint()
+	for _, addr := range maps.Keys(addressWeights) {
+		if cumWeight.GTE(threshold) {
+			break
 		}
 
-		executeData, err := types.CreateExecuteDataSinglesig(commandBatch.GetData(), commandBatchSig)
-		if err != nil {
-			return nil, nil, fmt.Errorf("could not create transaction data: %s", err)
+		sig, _ := sigs[addr]
+		if sig != nil {
```

Comment:
this should never happen. Remove and ensure it panics way before we get here? Call ValidateBasic on the request?

Reply:
Not all participants submitted a signature. Maybe add a comment to clarify this behaviour and difference between the keys of the two maps.

Reply:
oh yes, forgot to tell Chris

Reply:
then I prefer to not suppress the `ok` return and use that as a check instead